src/fusor/models.py

```diff
@@ -63,11 +60,12 @@ class FunctionalDomain(BaseModel):
 
     type: Literal[FUSORTypes.FUNCTIONAL_DOMAIN] = FUSORTypes.FUNCTIONAL_DOMAIN
     status: DomainStatus
-    associated_gene: GeneDescriptor
+    associated_gene: Gene
     id: CURIE | None = Field(None, alias="_id")
     label: StrictStr | None = None
-    sequence_location: LocationDescriptor | None = None
+    sequence_location: SequenceLocation | None = None
 
+    # TODO: is this obsolete now that vrsatile has been removed?
```

Comment:
help wanted

Reply:
is this referring to the field_validator? tbh I'm always confused by how these function but I believe this points to the `.id` property of the `FunctionalDomain` instance, it's just a thing we rig up to make CURIE validation work properly